makelib/xpkg.mk

```diff
@@ -0,0 +1,115 @@
+# Copyright 2022 The Upbound Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ====================================================================================
+# Options
+
+ifeq ($(origin XPKG_DIR),undefined)
+XPKG_DIR := $(ROOT_DIR)/package
+endif
+
+ifeq ($(origin XPKG_EXAMPLES_DIR),undefined)
+XPKG_EXAMPLES_DIR := $(ROOT_DIR)/examples
+endif
+
+ifeq ($(origin XPKG_IGNORE),undefined)
+XPKG_IGNORE := ''
+endif
+
+ifeq ($(origin XPKG_OUTPUT_DIR),undefined)
+XPKG_OUTPUT_DIR := $(OUTPUT_DIR)/xpkg
+endif
+
+# a registry that is scoped to the current build tree on this host. this enables
+# us to have isolation between concurrent builds on the same system, as in the case
+# of multiple working directories or on a CI system with multiple executors. All images
+# tagged with this build registry can safely be untagged/removed at the end of the build.
+ifeq ($(origin BUILD_REGISTRY), undefined)
+BUILD_REGISTRY := build-$(shell echo $(HOSTNAME)-$(ROOT_DIR) | shasum -a 256 | cut -c1-8)
+endif
+
+XPKG_REG_ORGS ?= xpkg.upbound.io/crossplane
+XPKG_REG_ORGS_NO_PROMOTE ?= xpkg.upbound.io/crossplane
+XPKG_LINUX_PLATFORMS := $(filter linux_%,$(PLATFORMS))
+XPKG_ARCHS := $(subst linux_,,$(filter linux_%,$(PLATFORMS)))
+XPKG_PLATFORMS := $(subst _,/,$(subst $(SPACE),$(COMMA),$(filter linux_%,$(PLATFORMS))))
+XPKG_PLATFORMS_LIST := $(subst _,/,$(filter linux_%,$(PLATFORMS)))
+XPKG_PLATFORM := $(subst _,/,$(PLATFORM))
+
+UP ?= up
```

Comment:
In some of the projects we've been delegating managing the up binary to the build submodule, do we want to do the same here or just use the `up` binary in the system's `PATH`?